pkgs/tools/graphics/gmic-qt/default.nix

```diff
@@ -77,18 +80,22 @@ stdenv.mkDerivation rec {
     curl
   ] ++ variants.${variant}.extraDeps or [];
 
-  cmakeFlags = [
-    "-DGMIC_QT_HOST=${if variant == "standalone" then "none" else variant}"
-    "-DENABLE_SYSTEM_GMIC=ON"
-    "-DENABLE_DYNAMIC_LINKING=ON"
-  ];
+  preConfigure = ''
+    cd gmic-qt
+  '';
```

Comment:
Actually, I have one more question. Why was `sourceRoot` removed in favor of this?

Afaik we shouldn't need anything from the root dir of the tarball, and I thought `sourceRoot` was the right way to indicate that everything will be happening from a subdir

Reply:
It is a bit subjective. 

`sourceRoot` is not reliable, because it depends on the previous `sourceRoot`, and it is opaquely determined by the fetcher.
Some like fetchurl create `${pname}-${version}`, others like fetchzip use `source`. And any future change on the internals of those fetchers can break this.

Indeed a similar thing occurred a short time ago: many `data/fonts` changed from using `fetchzip` internal machinery to using the good old `stdenv`, because `fetchzip` changed internally.

Because of it, I prefer to `cd` to the desired directory instead.

Reply:
That makes sense, but sucks a lot. I wish we had something that _did_ basically do a `cd` then without having to know what the fetcher does. Doing `cd` in `preConfigure` instead of `sourceRoot` will fail some builders (but I imagine `postPatch` should work pretty generally -- obviously that's semantically fraught though)

I guess that's out of scope for this PR though since this works 👍🏻